middleware/kubernetes/handler.go

```diff
@@ -23,6 +23,8 @@ func (k Kubernetes) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.M
 	m.Authoritative, m.RecursionAvailable, m.Compress = true, true, true
 
 	// TODO: find an alternative to this block
+	// TODO(miek): Why is this even here, why does the path Etcd takes not work?
```

Comment:
This PR is not finished yet (and k8s testing it borked right now). But it would be nice if the PTR lookup could also be moved to `middleware.PTR` as well. /cc @yongtang @johnbelamaric